scriptmodules/emulators/mame.sh

```diff
@@ -0,0 +1,157 @@
+}
+
+function install_mame() {
+    md_ret_files=(
+        'artwork'
+        'bgfx'
+        'ctrlr'
+        'docs'
+        'hash'
+        'hlsl'
+        'ini'
+        'language'
+        "$(_get_binary_name_${md_id})"
+        'nl_examples'
+        'plugins'
+        'roms'
+        'samples'
+        'uismall.bdf'
+        'LICENSE.md'
+    )
+}
+
+function configure_mame() {
+    local system="mame"
+    mkRomDir "arcade"
+    mkRomDir "$system"
+
+    moveConfigDir "$home/.mame" "$md_conf_root/$system"
+
+    # Create required MAME directories underneath the ROM directory
+    if [[ "$md_mode" == "install" ]]; then
+        local mame_sub_dir
+        for mame_sub_dir in artwork cfg comments diff inp nvram samples scores snap sta; do
+            mkRomDir "$system/$mame_sub_dir"
+        done
+    fi
+
+    # Create a BIOS directory, where people will be able to store their BIOS files, separate from ROMs
+    mkUserDir "$biosdir/$system"
+    chown $user:$user "$biosdir/$system"
+
+    # Create a new INI file if one does not already exist
+    if [[ "$md_mode" == "install" && ! -f "$md_conf_root/$system/mame.ini" ]]; then
+        pushd "$md_conf_root/$system/"
+        "$md_inst/$(_get_binary_name_${md_id})" -createconfig
+        popd
+
+        iniConfig " " "" "$md_conf_root/$system/mame.ini"
+        iniSet "rompath"            "$romdir/$system;$romdir/arcade;$biosdir/$system"
+        iniSet "hashpath"           "$md_inst/hash"
+        iniSet "samplepath"         "$romdir/$system/samples;$romdir/arcade/samples"
+        iniSet "artpath"            "$romdir/$system/artwork;$romdir/arcade/artwork"
+        iniSet "ctrlrpath"          "$md_inst/ctrlr"
+        iniSet "pluginspath"        "$md_inst/plugins"
+        iniSet "languagepath"       "$md_inst/language"
+
+        iniSet "cfg_directory"      "$romdir/$system/cfg"
+        iniSet "nvram_directory"    "$romdir/$system/nvram"
+        iniSet "input_directory"    "$romdir/$system/inp"
+        iniSet "state_directory"    "$romdir/$system/sta"
+        iniSet "snapshot_directory" "$romdir/$system/snap"
+        iniSet "diff_directory"     "$romdir/$system/diff"
+        iniSet "comment_directory"  "$romdir/$system/comments"
+
+        iniSet "skip_gameinfo" "1"
+        iniSet "plugin" "hiscore"
+        iniSet "samplerate" "44100"
+
+        iniConfig " " "" "$md_conf_root/$system/ui.ini"
+        iniSet "scores_directory" "$romdir/$system/scores"
+
+        iniConfig " " "" "$md_conf_root/$system/plugin.ini"
+        iniSet "hiscore" "1"
+
+        iniConfig " " "" "$md_conf_root/$system/hiscore.ini"
+        iniSet "hi_path" "$romdir/$system/scores"
+
+        chown -R $user:$user "$md_conf_root/$system"
+        chmod a+r "$md_conf_root/$system/mame.ini"
+    fi
+
+    local binary_name="$(_get_binary_name_${md_id})"
+    addEmulator 0 "$md_id" "arcade" "$md_inst/${binary_name} %BASENAME%"
+    addEmulator 1 "$md_id" "$system" "$md_inst/${binary_name} %BASENAME%"
```

Comment:
i think this wants to be `addEmulator 0` as we don't want it to be the default MAME emulator